File: Unidad 3-Funciones/Ejercicio22.js
/*
Autor: Jose Eduardo Mendez Verdejo
Problema:.Determina el valor de la secuencia
1! + 2! + .. +n!
Entrada: un número entero
Salida: la suma de factoriales desde 1 hasta el número seleccionado
Proceso:Introducir un numero para ir sacando el total del factorial de mayor a menor sumando los resultados en una variable he imprimir el resultado.
*/

var numero = entrada();
var i = 1;
var suma = 0;

while(i <= numero){
	suma = suma + proceso(i);
    i = i + 1;
}
salida(suma);


function entrada(){
	var numero = parseInt(prompt("Ingrese un numero: "));
  while(numero < 1){
    var numero = parseInt(prompt("Ingrese un numero: "));
  }
	return numero;
}

function proceso(i){
	var factorial = 1, j = 1;
	for(j; j <= i; j++){
		factorial = factorial * j;
	}
	return factorial;
}

function salida(numero){
	alert("La suma de los factoriales es:" + numero);
}

/*
QA: Raul Rivadeneyra

entrada():
Todos los valores numericos ingresados funcionan, OK

proceso():
Calcula correctamente el factorial de i, OK

salida():
Muestra correctamente el dato, Ok

Prueba integral:
Correctamente realiza su tarea el codigo, con las validaciones necesarias, 10/10
*/

File: Unidad 3-Funciones/Ejercicio22.py
#Autor:Deyberth Carrillo
#Entrada: Un numero n
#Salida: La suma de 1! hasta n!

#Funciones
def entrada_N():
    entrada=int(input())
    return entrada
def factorial(num):
    factorialNum=1
    for i in range(1,num+1):
        factorialNum=factorialNum*i
    return factorialNum
def imprimirSuma(sumaFactoriales):
    print(sumaFactoriales)
    
#Main

#Inicialización de variables
n=0
sumaFactorial = 0
#Entrada
n=entrada_N()

#Proceso
#Se calcula la suma de 1! hasta n!
for i in range(1,n+1):
    sumaFactorial += factorial(i)
#Salida
imprimirSuma(sumaFactorial)
